api/types/access_request.go

```diff
@@ -642,6 +642,26 @@ func (u *AccessRequestUpdate) Check() error {
 	return nil
 }
 
+// RequestReasonMode can be either "required" or "optional". Empty-string is treated as "optional".
+// If a role has the request reason mode set to "required", then reason is required for all Access
+// Requests requesting roles or resources allowed by this role. It applies only to users who have
+// this role assigned.
+type RequestReasonMode string
+
+const (
+	RequestReasonModeRequired RequestReasonMode = "required"
+	RequestReasonModeOptional RequestReasonMode = "optional"
+)
+
+func (m RequestReasonMode) Required() bool {
```

Comment:
I would add a couple of comments to those public consts and the method.
Also, what Tiago said.

Reply:
https://github.com/gravitational/teleport/compare/2a31d2df467bf13941e9dd905d52ec4c4507fe12..a9c62defa98a6c9e17f3e961c9264b44b145c113